### PR TITLE
Enable writings node.js buffers directly

### PR DESCRIPTION
**Pull request by EricMCornelius**
Hi there,

I ran into encoding problems with binary request string data (gzipped payloads) being transformed via Buffer.toString('utf8') in transformRequest.

This PR will enable passing in node buffers directly instead of needing to create a stream or copy to an ArrayBuffer just for the sake of sending compressed data.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -73,6 +73,6 @@
   },
   "typings": "./index.d.ts",
   "dependencies": {
-    "follow-redirects": "1.0.0"
+    "follow-redirects": "0.0.7"
```
Comment (nickuraltsev) [inline package.json]: Why is this necessary?

**Comment by EricMCornelius** [comment]
Sorry, I believe the downgrade was because I ran into a serious regression w/ the updated follow-redirects dependency and updated my master branch of my fork.

Using the updated follow-redirects causes this reported write after end issue: https://github.com/olalonde/follow-redirects/issues/50 - which I was hitting in the component using my axios fork.  I haven't had time to investigate that dependency defect further.

**Comment by EricMCornelius** [comment]
https://github.com/mzabriskie/axios/issues/623 - I see there's already some awareness of the problem with follow-redirects as well.

**Comment by jcready** [comment]
I'm also running into the problem with follow-redirects. Setting `maxRedirects: 0` fixes the problem, but is obviously not ideal.

**Comment by jcready** [comment]
It looks like the problem with follow-redirects was [fixed in a later minor version](https://github.com/olalonde/follow-redirects/commit/9eec6f0fb8d1c51dbd9be732ce1e2c794a01b652) but axios locks down the version to 1.0.0.

**Comment by rubennorte** [comment]
Sorry but we've finally merged #773 as it seemed more complete. Thanks for your contribution.

**Comment by jcready** [comment]
@rubennorte what about the issue with follow-redirects? Should that be opened as a separate PR?

**Comment by rubennorte** [comment]
@jcready yes please
